Sources/TuistKit/Services/ShareService.swift

```diff
@@ -226,6 +234,57 @@ struct ShareService {
         }
     }
 
+    private func shareIPA(
+        _ appPaths: [AbsolutePath],
+        fullHandle: String,
+        serverURL: URL
+    ) async throws {
+        guard appPaths.count == 1,
+              let ipaPath = appPaths.first else { throw ShareServiceError.multipleAppsSpecified(appPaths.map(\.pathString)) }
+
+        guard let appBundlePath = try fileArchiverFactory.makeFileUnarchiver(for: ipaPath).unzip().glob("**/*.app").first
+        else { throw ShareServiceError.appBundleInIPANotFound(ipaPath) }
+        let appBundle = try await appBundleLoader.load(appBundlePath)
+        let displayName = appBundle.infoPlist.name
+
+        let preview = try await previewsUploadService.uploadPreviews(
+            .ipa(ipaPath),
+            displayName: displayName,
+            version: appBundle.infoPlist.version.description,
+            bundleIdentifier: appBundle.infoPlist.bundleId,
+            fullHandle: fullHandle,
+            serverURL: serverURL
+        )
+        logger
+            .notice(
+                "\(displayName) uploaded – share it with others using the following link: \(preview.url.absoluteString)"
+            )
+    }
+
+    private func shareAppBundles(
+        _ appPaths: [AbsolutePath],
+        fullHandle: String,
+        serverURL: URL
+    ) async throws {
+        let appBundles = try await appPaths.concurrentMap {
+            try await appBundleLoader.load($0)
+        }
+
+        let appNames = appBundles.map(\.infoPlist.name).uniqued()
+        guard appNames.count == 1,
+              let appName = appNames.first else { throw ShareServiceError.multipleAppsSpecified(appNames) }
+
+        let preview = try await previewsUploadService.uploadPreviews(
```

Comment:
Similar comment here regarding the indication of "something asynchronous is going to start"